foundations/src/telemetry/settings/tracing.rs

```diff
@@ -26,6 +26,13 @@ pub struct TracingSettings {
 
     /// The strategy used to sample traces.
     pub sampling_strategy: SamplingStrategy,
+    /// Enable liveness tracking of all generated spans. Even if the spans are
+    /// unsampled. This can be useful for debugging potential hangs cause by
+    /// some objects remaining in memory.  The default value is false, meaning
+    /// only sampled spans are tracked.
+    ///
+    /// To get a json dump of the currently active spans, query: `/debug/traces`
+    pub track_all_spans: bool,
```

Comment:
I think we need to make the whole liveness tracking functionality, i.e. introduce `LivenessTrackingSettings` with `enabled` field and also put this field in there too.